inter-ident-verif/README.md

```diff
@@ -0,0 +1,62 @@
+3. User encrypts result with Apple public key, preventing the caller from decrypting to get the original message.
+4. User sends result to Apple Tech Support.
+
+### Apple Tech Support
+5. Apple Tech Support encrypts message with Apple Tech Support private key, ensuring that Apple knows it's from them.
+6. Apple Tech Support sends result to Apple.
+
+### Apple
+7. Apple recognizes Apple Tech Support as valid, so proceeds.
+8. Apple decrypts message with Apple Tech Support public key.
+9. Apple decrypts result with Apple private key.
+10. Apple decrypts result with user public key, which was previously protected by Apple RSA key. Apple now has the original message.
+
+11. Apple encrypts result with Apple private key, acting as a signature from Apple directly.
+12. Apple encrypts result with user public key, preventing the caller from decrypting the original message.
+13. Apple encrypts result with Apple Tech Support public key, verifying validity of trusted caller.
+14. Apple sends result to Apple Tech Support.
+
+### Apple Tech Support
+15. Apple Tech Support decrypts message with Apple Tech Support private key, and can't decrypted further.
+16. Apple Tech Support sends result to user.
+
+### User
+17. User decrypts result with user private key.
+18. User decrypts message with Apple public key.
+19. User verifies that result is the same as the original message generated.
+
+Following this algorithm, the caller can never decrypt the message being sent between the user and the main company, and the main company verifies and signs the users unique message.
+
+The key aspect here is that the message being sent and received by the user is the same, but encrypted in a different order. This means that it's essential to use a non-communitive cryptosystem.
+___
+## Readable hashes
+An important aspect of this process is that users are able to read their hashes over the phone. To increase the readability of these hashes, I propose a modified number representation system similar to hexadecimal where each digit has 64 variants, `{a-z, A-Z, 0-9, *, #}`, allowing 6-bits of information to be stored per digit. With only 6 digits in this number system, we can represent up to 68,719,476,736, covering all 32-bit numbers. With 11 digits, we can represent up to 7.38e19, covering all 64-bit numbers. For shorter messages, this should be enough to read these values over voice.
+___
+## User interface
+When the user receives a call, they ask for the callers root company. They then navigate and log into the Kanidm web interface on their smartphone, search up the root company, and click a button that does steps 1-3, and displays the result to the user in the number system specified above. This code is read over the phone, where 5-15 are performed remotely for the user. Once the caller reads the resulting message back to the user, they will enter this back into the web interface, which performs steps 17-19, and tells the user if the verification was successful.
```

Comment:
This assumes the user has an account on the kanidm instance at the company, yes? 

It's not clear here how the "read numbers" fit in with steps 17-19 or steps 5-15. 

Additionally, it is worth asking how this defends from replay attacks? I assume the message in this case is actually a nonce?

Reply:
I don't know how to respond to comments with quoted code so I made responses below detailing how I addressed things you brought up. Thanks for the feedback!